reconcile/utils/terrascript_aws_client.py

```diff
@@ -224,12 +240,23 @@ def __init__(self, integration: str, integration_prefix: str,
                 version="2.2.0"
             )
 
-            b = Backend("s3",
-                        access_key=config['aws_access_key_id'],
-                        secret_key=config['aws_secret_access_key'],
-                        bucket=config['bucket'],
-                        key=config['{}_key'.format(integration)],
-                        region=config['region'])
+            for account in filtered_accounts:
+                terraform_state = account.get("terraform_state")
+                if not terraform_state:
+                    b = Backend("s3",
+                                access_key=config['aws_access_key_id'],
+                                secret_key=config['aws_secret_access_key'],
+                                bucket=config['bucket'],
+                                key=config['{}_key'.format(integration)],
+                                region=config['region'])
+                else:
+                    integration = terraform_state['integrations']['integration']
```

Comment:
`terraform_state["integrations"]` is a list so you need to find the one that matches the currently executed integration provided as the `integration` variable to this function.